frontend/src/pages/CustomDrink/CustomDrink.tsx

```diff
@@ -0,0 +1,135 @@
+import "./CustomDrink.css"
+import {FormEvent, useState} from "react";
+import {postCustomDrink, sendPicture} from "../../apiServices/service";
+import {CustomIngredientModel} from "../../components/Model";
+import {useNavigate} from "react-router-dom";
+
+
+export default function CustomDrink(){
+
+
+    const[instruction,setInstruction] = useState("")
+    const[ingredients, setIngredients] = useState<CustomIngredientModel[]>([])
+    const[ingredientName,setIngredientName] = useState("")
+    const[amount,setAmount] = useState(0)
+    const[unit, setUnit] = useState("")
+    const[glass, setGlass] = useState("")
+    const[image,setImage] = useState({} as File)
+    const[cocktailName, setCocktailName] = useState("")
+    const nav = useNavigate()
+
+
+
+    const addCustomDrink = (event:FormEvent) => {
+        event.preventDefault()
+        handleUpload()
+
+
+
+    }
+
+    const handleUpload = () => {
+        const formData = new FormData()
+        formData.append("file",image)
+        formData.append("upload_preset","customDrink")
+        sendPicture(formData)
+            .then((data) => {
+                const currentDrink =  {
+                    customDrinkName:cocktailName,
+                    customDrinkURL:data.secure_url,
+                    customIngredients:ingredients,
+                    customInstruction:instruction,
+                    customGlass:glass
+
+                }
+                postCustomDrink(currentDrink)
+                    .then(()=> {
+                        setCocktailName("")
+                        setImage({} as File)
+                        setInstruction("")
+                        setAmount(0)
+                        setUnit("")
+                        setIngredientName("")
+                        setGlass("")
+                        alert("Your drink has just been added!")
+                    })
+                    .catch(()=> {localStorage.removeItem("jwt")
```

Comment:
Maybe you should check if it really was a 403 Statuscode, before you logout the user, the could be another error.